python/pyspark/sql/tests/test_udtf.py

```diff
@@ -256,29 +254,76 @@ def eval(self, a: int):
         ):
             TestUDTF(lit(1), lit(2)).collect()
 
+    def test_udtf_init_with_additional_args(self):
+        @udtf(returnType="x int")
+        class TestUDTF:
+            def __init__(self, a: int):
+                ...
+
+            def eval(self, a: int):
+                yield a,
+
+        with self.assertRaisesRegex(
+            PythonException, r"__init__\(\) missing 1 required positional argument: 'a'"
```

Comment:
I mentioned this in @ueshin's previous PR, it would be helpful to include the UDTF name in the error message. Is it possible to add that now, or should we separate that work to a later PR?

Reply:
Yup here is the Jira ticket to add UDTF name in the error message: https://issues.apache.org/jira/browse/SPARK-44008